sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -69,6 +70,23 @@ private[hive] case class HiveSimpleUDF(
     udfType != null && udfType.deterministic() && !udfType.stateful()
   }
 
+  override def inputTypes: Seq[AbstractDataType] = {
+    val inTypes = children.map(_.dataType)
+    if (!inTypes.exists(_.existsRecursively(_.isInstanceOf[DecimalType]))) {
```

Comment:
Why do we need special handling for decimal types?

Reply:
It's a compatible issue. In normal case, data type is converted by Hive ObjectInspector at running time. But Hive not support input decimal type when method required double type. Unfortunately the default type of `1.1` is different between Spark and Hive, which are decimal and double. Then caused this issue.

Reply:
Ah, I see. Could you describe it in the PR description?

Reply:
Yeah.

Reply:
Sorry I didn't get it. `ImplicitCastInputTypes` can implicit cast decimal to double, doesn't it?

Reply:
The issue is for decimal to double, so I only check the decimal type and skip other type at here.

Reply:
what's the drawback if we do it for all other types?

Reply:
In first commit I did it for all types but test not passed. The reason is a UDF required an Object type.

We convert Java Object Type to NullType in `HiveInspectors.javaTypeToDataType` (seems this can be changed ?)
```
 // Hive seems to return this for struct types?
    case c: Class[_] if c == classOf[java.lang.Object] => NullType
```

So we can't reflect the UDF method using a NullType, the error msg is:
```
in query: cannot resolve 'example_format('%o', 93)' due to data type mismatch: argument 2 requires array<null> type, however, '93' is of int type.; line 1 pos 7;
```

Reply:
if a hive udf requires `Object`, I think it means `AnyDataType`. We should only special case it.

Reply:
How about this ? 
```
    val expectTypes = method.getGenericParameterTypes.map(javaTypeToDataType)
    if (expectTypes.exists(_.existsRecursively(_.isInstanceOf[NullType]))) {
      children.map(_.dataType)
    } else {
      expectTypes
    }
```

Reply:
method.getGenericParameterTypes.map(javaTypeToDataType).map { dt =>
  if (dt.existsRecursively(_.isInstanceOf[NullType])) AnyDataType else dt
}

Reply:
Seems we should check data type and replace `NullType` to `AnyDataType` on by one to avoid such case `Map<Double, Object> input`.

I will do an another check if we can change `HiveInspectors.javaTypeToDataType` that using `AnyDataType` directly.